packages/squiggle-lang/src/js/index.ts

```diff
@@ -50,6 +50,67 @@ export let defaultSamplingInputs: samplingParams = {
   xyPointLength: 10000,
 };
 
+import type { t as SampleSetDist_t } from "../rescript/Distributions/SampleSetDist/SampleSetDist.gen";
+import type { mixedShape } from "../rescript/Distributions/PointSetDist/PointSetTypes.gen";
+
+import type { symbolicDist as SymbolicDistTypes_symbolicDist } from "../rescript/Distributions/SymbolicDist/SymbolicDistTypes.gen";
+
+type rescriptPointSetDist =
+  | {
+      TAG: 0; // Mixed
+      _0: mixedShape;
+    }
+  | {
+      TAG: 1; // Discrete
+      _0: discreteShape;
+    }
+  | {
+      TAG: 2; // ContinuousShape
+      _0: continuousShape;
+    };
```

Comment:
It could be good to convert this into something more readable. Like,
```
{pointSetType: "Mixed", shape: shape}
```

It would be a bit slower, but would be more readable. I don't think I care about performance too much for this specific aspect.

Reply:
I don't have liberty over these type declarations. I'm just describing here what's exported by rescript. However, in terms of the final exported values, it could be possible to move from `{tag: "Mixed", value: continuousShape}` to `{pointSetType: "Mixed", shape: continuousShape}`?

Reply:
Can't we write some code with Rescript to output a format that we prefer? It would be more work, for sure, but I think it's possible. We could just a JSON formatter or something.

Reply:
We could. The really unreadable format of `{TAG: number, _0: value}` comes up whenever you use any variant, so if you wanted to convert it to something prettier, you could write some rescript code that converts it into a format that doesn't have any variants at all in it. This obviously has the problem that writing sum types in records is also very dirty. It would be quite a bit of work to go down the entire tree just for the sake of an intermediary representation that never gets shown to the user anyway.

Reply:
>  It would be quite a bit of work to go down the entire tree just for the sake of an intermediary representation that never gets shown to the user anyway.

My guess is that we might eventually want to bring up much of this for the user, but not all. That said, that could be 3-20 months down.